From 4bbc7a2f5c77921e687f33e84e79776859eddfdb Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Tue, 16 Jan 2018 18:57:31 +0100
Subject: [PATCH] Fix build with >=attr-2.4.48

Summary: It was looking for long deprecated attr/xattr.h header.

Test Plan: Built fine with attr-2.4.47 and 2.4.48.

Reviewers: #frameworks, mgallien

Reviewed By: mgallien

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D9880
---
 cmake/FindXattr.cmake | 2 +-
 src/xattr_p.h         | 4 +---
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/cmake/FindXattr.cmake b/cmake/FindXattr.cmake
index 3004061..a33f316 100644
--- a/cmake/FindXattr.cmake
+++ b/cmake/FindXattr.cmake
@@ -13,7 +13,7 @@ if (XATTR_INCLUDE_DIRS AND XATTR_LIBRARIES)
     set(XATTR_FOUND TRUE)
 else ()
 
-    find_path(XATTR_INCLUDE attr/xattr.h
+    find_path(XATTR_INCLUDE sys/xattr.h
         /usr/include
         /usr/local/include
         ${CMAKE_INCLUDE_PATH}
diff --git a/src/xattr_p.h b/src/xattr_p.h
index 4663347..542c021 100644
--- a/src/xattr_p.h
+++ b/src/xattr_p.h
@@ -32,13 +32,11 @@
 #include <sys/types.h>
 #include <sys/xattr.h>
 
-#if defined(Q_OS_ANDROID)
+#if defined(Q_OS_ANDROID) || defined(Q_OS_LINUX)
 // attr/xattr.h is not available in the Android NDK so we are defining ENOATTR ourself
 #ifndef ENOATTR
 # define ENOATTR ENODATA        /* No such attribute */
 #endif
-#else
-#include <attr/xattr.h>
 #endif
 
 #include <errno.h>
